### PR TITLE
Extend ADM Deployment component tests (Part 2)

### PR DESCRIPTION
**Pull request by pudelkoM**
## Description

Describe the purpose of this pull request.

## Changes

List the changes you have made.

## Additional Information

Include any additional information, such as how to test your changes.

## Checklist

- [ ] Tests passed
- [ ] Documentation updated

### CODE REVIEW

File: app-deployment-manager/test/deployment_delete/deployment_delete_test.go [hunk 1]
```diff
@@ -65,3 +66,26 @@ func (s *TestSuite) TestDeleteDeploymentParentOnlyDeleteType() {
 
 	s.T().Logf("successfully deleted deployment with ID: %s using delete type", deploymentID)
 }
+
+func (s *TestSuite) TestDeleteDeploymentAllDeleteType() {
```
Comment (adibrastegarnia) [inline app-deployment-manager/test/deployment_delete/deployment_delete_test.go:70]: That reason that I didn't add this because we need to see how we can check the actual behavior of this to distinguish it with other case PARENT_ONLY. My understanding is that this deletes all deployments in all targets based on this code https://github.com/open-edge-platform/app-orch-deployment/blob/main/app-deployment-manager/internal/northbound/northbound.go#L1148

File: app-deployment-manager/test/deployment_update/deployment_update_test.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+	}
+	deployID, code, err := deploymentutils.StartDeployment(deploymentReq)
+	s.Equal(http.StatusOK, code)
+	s.NoError(err, "Failed to create '"+deploymentutils.AppNginx+"-"+deploymentutils.DeploymentTypeTargeted+"' deployment")
+
+	deployment, code, err := deploymentutils.GetDeployment(s.AdmClient, deployID)
+	s.Equal(http.StatusOK, code, "Expected HTTP status 200 for getting deployment details")
+	s.NoError(err, "Failed to get deployment details")
+
+	// TODO: some modification the deployment object to update?
```
Comment (adibrastegarnia) [inline app-deployment-manager/test/deployment_update/deployment_update_test.go:32]: I think we should not postpone this as TODO and we should try to find a valid update. for example you can use wordpress to upgrade the app version and this is the main goal of this test to see if Upgrade works or not.

File: app-deployment-manager/test/deployment_list/deployment_list_test.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+			DeploymentTimeout: deploymentutils.DeploymentTimeout,
+			DeleteTimeout:     deploymentutils.DeleteTimeout,
+			TestName:          testName,
+		}
+		_, code, err := deploymentutils.StartDeployment(deploymentReq)
+		s.Equal(http.StatusOK, code)
+		s.NoError(err, "Failed to create '"+app+"-"+deploymentutils.DeploymentTypeTargeted+"' deployment")
+	}
+
+	deps, code, err := deploymentutils.DeploymentsListWithParams(s.AdmClient, &restClient.DeploymentServiceListDeploymentsParams{
```
Comment (adibrastegarnia) [inline app-deployment-manager/test/deployment_list/deployment_list_test.go:35]: Can we extend this test a little bit or add new tests to test pageSize and offset in out of range or negative values? if it makes sense we should add new tests for invalid use cases if we don't have one.

**Review by adibrastegarnia — CHANGES_REQUESTED** [review]

File: app-deployment-manager/test/deployment_delete/deployment_delete_test.go (hunk 1 above)
Comment (pudelkoM) [reply app-deployment-manager/test/deployment_delete/deployment_delete_test.go:70]: I see. I think we want to trigger case 3 here (https://github.com/open-edge-platform/app-orch-deployment/blob/main/app-deployment-manager/internal/northbound/northbound.go#L1202).

File: app-deployment-manager/test/deployment_list/deployment_list_test.go (hunk 3 above)
Comment (pudelkoM) [reply app-deployment-manager/test/deployment_list/deployment_list_test.go]: done

**Review by nathsudi — APPROVED** [review]
